### PR TITLE
App with Ruby on Rails blog post

### PR DESCRIPTION
**Pull request by unaitxu**

### CODE REVIEW

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+{% imagebasic blog/blog-sales-app-dashboard.png %} image {% endimagebasic %}
+
+The main purpose of the Dashboard was to have all the basic information about a shop at a glance.
+So we decided to include a graph of sales, the amount of sales and orders, as well as the top products, and last orders.
+
+Then we included three charts on the Sales section.
+The first is the same as in the Dashboard, the second one represents the orders on the selected time frame and the last is the revenue per order, the calculation between the first two graphs.
+We also added the option to export the data to an `xlsx` file so the merchant doesn't need an internet connection to review the sales.
+
+The Customers section is a simple table including all the merchant customers ordered via number of orders or the total amount spent.
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: In order to easily capture, what's meant, I suggest to include screenshots for the sections, you're writing about, like Products, Products Detail, Customers, Sales.

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
+---
+layout: post
+title: "Creating a Sales Analytics App with Ruby on Rails"
+date: "2017-02-20 07:00:00"
+image: blog-header/hamburg-port-rails.jpg
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: I like the header image very much but on the blog detail page you don't see the bridge anymore.
Maybe one of those fit better:
https://www.pexels.com/photo/rails-train-path-straight-1425/
https://www.pexels.com/photo/sky-forest-rails-straight-981/
https://www.pexels.com/photo/mountains-nature-sunny-bridge-1018/

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+---
+
+One of the languages that we use the most on the ePages Tools Team is [Ruby](https://www.ruby-lang.org).
+We love the syntax, the simplicity and the big and active community around it.
+Some time ago, [we wrote about our SDK](https://developer.epages.com/blog/2016/02/02/on-the-way-to-a-ruby-sdk.html) for the ePages shops and lots of things have changed since then.
+We've kept developing it to support more operations, such as data pagination and also to keep up with the releases of our colleagues from the [API development team](https://developer.epages.com/apps).
+
+## Status of the SDK
+
+We are preparing a release Soon&#8482; containing lots of model changes, in order to make the use of [the gem](https://github.com/ePages-de/epages-rest-ruby) even easier.
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: what's this: Soon&#8482; ?
Maybe I don't understand this "Soon TM" thingy. Let's talk about it.

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+In every project you have to make some decisions on what tech you use.
+We decided to use [Rails](http://rubyonrails.org/) - in part because we love it, but also to use our [own developed library](https://github.com/ePages-de/epages-rest-ruby).
+On the charts side there are lots of options to choose, but we went with [D3](https://d3js.org/), an open source, very powerful and very customizable option.
+The main reason of picking this was purely educational.
+I love learning new stuff, libraries, languages, paradigms, you name it; and I already used [Highcharts](http://www.highcharts.com/), [Google charts](https://developers.google.com/chart/) and [Chart.js](http://www.chartjs.org/), plus I was really curious about it and my team let me decide {% emoji smiley %}.
+
+[Devise](https://github.com/plataformatec/devise) was our option for user management.
+This gem is easy and simple, so it was perfect to deliver fast (even though in the end everything became more complicated).
+For this simplicity reason we choose [Slim](http://slim-lang.com/), [Sass](http://sass-lang.com/) and [CoffeeScript](http://coffeescript.org/).
+
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: As this post is quite long, I suggest to make two posts out of it and stop the first one here.
Maybe find some nice teaser to add here for the next one.

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+## Issues on the development process
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Start from here for another post.
Need to find a nice headline, image and intro. I can help you out with that.
Once the second one is in place, I'll have another check on it.

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 6]
```diff
@@ -0,0 +1,173 @@
+---
+layout: post
+title: "Creating a Sales Analytics App with Ruby on Rails"
+date: "2017-02-20 07:00:00"
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: We can talk about a publication date once the review is done. Depending on when you can make it, I suggest either the 21st or 23rd of February.

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 2 above)
Comment (unaitxu) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: I liked the fact that it were Hamburg Rails. But if you say it's better to change, I trust you, so just choose the image that you think it fits the best

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 3 above)
Comment (unaitxu) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: It's a joke http://www.urbandictionary.com/define.php?term=soon%E2%84%A2 but if you think it's not nice to include this things, we can just change it to a simple soon

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 1 above)
Comment (unaitxu) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: 👍  will do

File: _posts/2017/2017-03-10-creating-a-sales-analytics-app-with-ruby-on-rails-2.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
+---
```
Comment (BirgitBader) [inline _posts/2017/2017-03-10-creating-a-sales-analytics-app-with-ruby-on-rails-2.md]: Please make a separate Pull Request for this post.

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+For simplicity we mimicked more or less the API models and added a bit of Rails Magic so it's even easier to use.
+
+{% image blog/blog-sales-app-model.png auto center %}
+
+After that we created all the sections of the App: Dashboard, Sales, Products and Customers. All these parts are bounded to two dates (The start date and end date) so the merchant can get even more useful data faster.
+
+The main purpose of Dashboard was to have all the basic information about your shop in a glance, so we decided to include a graph of sales, the amount of sales and orders, as well as the top products and last orders.
+
+Then we included three charts on the Sales section, the first is the same as in the Dashboard, the second one represents the orders on the selected time frame and the last is the revenue per order, the calculation between the first two graphs. We also added the option to export the data to an `xlsx` file so the Merchant doesn't need and internet connection to review the sales.
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Please enter screenshots for the sections.

**Comment by BirgitBader** [comment]
@unaitxu somehow my committed changes are not included anymore, see commit: 8ed0d2180d338204099a2909cb15e00c3611fc98

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 9]
```diff
@@ -0,0 +1,109 @@
+---
+layout: post
+title: "Creating a Sales Analytics App with Ruby on Rails"
+date: "2017-02-20 07:00:00"
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Change date to 2017-03-02 
And change date in filename accordingly.

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+image: blog-header/hamburg-port-rails.jpg
+categories: tech-stories
+authors: ["Unai M."]
+---
+
+One of the languages that we use the most on the ePages Tools Team is [Ruby](https://www.ruby-lang.org).
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: One of the languages that we use the most in my team is...

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+    @page = 2,
+    @results = 42,
+    @results_per_page = 10>
+{% endhighlight %}
+
+so you have more control on the call and it gets easier to do batch operations.
+We also fixed some bugs and updated some dependencies.
+
+Testing things in the console was fun, but we needed a real life scenario to check that the gem was working correctly and to find missing features.
+Lucky for us, our Product Owner came up with task perfect for this situation.
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Lucky for us, our Product Owner came up with a perfect task for this.

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+We had mockups, designs and texts for the whole Web App, so we could focus on programming.
+
+For simplicity we mimicked more or less the API models and added a bit of Rails Magic so it's even easier to use.
+
+{% image blog/blog-sales-app-model.png auto center %}
+
+After that, we created all the sections of the app: Dashboard, Sales, Products, and Customers.
+All these parts are bound to two dates (start date and end date) so the merchant can get even more useful data faster.
+
+{% imagebasic blog/blog-sales-app-dashboard.png %} image {% endimagebasic %}
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Is it possible to remove your email address from the screenshot and fake it with "example@example.com"?

Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Include image caption for images.
Use here: Dashboard of Sales Analytics App

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+
+The main purpose of the Dashboard was to have all the basic information about a shop at a glance.
+So we decided to include a graph of sales, the amount of sales and orders, as well as the top products, and last orders.
+
+{% imagebasic blog/blog-sales-app-sales.png %} image {% endimagebasic %}
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Include image caption:
Use here: Sales view

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+
+Then we included three charts on the Sales section.
+The first is the same as in the Dashboard, the second one represents the orders on the selected time frame and the last is the revenue per order, the calculation between the first two graphs.
+We also added the option to export the data to an `xlsx` file so the merchant doesn't need an internet connection to review the sales.
+
+{% imagebasic blog/blog-sales-app-customers.png %} image {% endimagebasic %}
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Include image caption:
Use here: Customers view

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+
+The Customers section is a simple table including all the merchant customers ordered via number of orders or the total amount spent.
+Above the table we can see all the customers for the timeframe and if we had any returning ones.
+
+{% imagebasic blog/blog-sales-app-products.png %} image {% endimagebasic %}
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Include image caption:
Use here: Products view

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+
+{% imagebasic blog/blog-sales-app-sales.png %} image {% endimagebasic %}
+
+Then we included three charts on the Sales section.
+The first is the same as in the Dashboard, the second one represents the orders on the selected time frame and the last is the revenue per order, the calculation between the first two graphs.
+We also added the option to export the data to an `xlsx` file so the merchant doesn't need an internet connection to review the sales.
+
+{% imagebasic blog/blog-sales-app-customers.png %} image {% endimagebasic %}
+
+The Customers section is a simple table including all the merchant customers ordered via number of orders or the total amount spent.
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: The Customers section is a simple table including all the merchant's customers ordered by number of orders or the total amount spent.

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+Above the table we can see all the customers for the timeframe and if we had any returning ones.
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Above the table we can see all the customers for the specified time frame and and the number of returning customers.

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+
+{% imagebasic blog/blog-sales-app-products.png %} image {% endimagebasic %}
+
+The last part are the Products.
+In this view, we have the products ordered via revenue or amount of units sold.
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: In this view, we have the products ordered by revenue or amount of units sold.

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 19]
```diff
@@ -0,0 +1,109 @@
...
+Here we implemented the functionality of having more insights on the product.
+When clicking on one, you can see the extra data or cross selling products, this is, what other items have shared orders with.
+This allows the merchant to maybe realize on connections (or disconnections) between products in an easy and fast way.
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: ... in a quick and easy way.

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 20]
```diff
@@ -0,0 +1,109 @@
...
+
+The Customers section is a simple table including all the merchant customers ordered via number of orders or the total amount spent.
+Above the table we can see all the customers for the timeframe and if we had any returning ones.
+
+{% imagebasic blog/blog-sales-app-products.png %} image {% endimagebasic %}
+
+The last part are the Products.
+In this view, we have the products ordered via revenue or amount of units sold.
+Here we implemented the functionality of having more insights on the product.
+When clicking on one, you can see the extra data or cross selling products, this is, what other items have shared orders with.
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: By clicking on a product, the product details appear with additional information such as cross-selling products, i.e. what other items have shared orders with.

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 21]
```diff
@@ -0,0 +1,109 @@
...
+This allows the merchant to maybe realize on connections (or disconnections) between products in an easy and fast way.
+
+Both, the Customers and Products section, have the option to download the data on a `csv`.
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: ... `csv` file.

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 22]
```diff
@@ -0,0 +1,109 @@
...
+Other basic functionalities like changing you email address or password and 'Forgot my password' are also included in the app.
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Other basic functionalities like changing the email address or password as well as a 'Forgot my password' option are also included in the app.

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 23]
```diff
@@ -0,0 +1,109 @@
...
+
+## Some decisions we made
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Some decisions on technologies

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 24]
```diff
@@ -0,0 +1,109 @@
...
+
+In every project you have to make some decisions on what tech you use.
+We decided to use [Rails](http://rubyonrails.org/) - in part because we love it, but also to use our [own developed library](https://github.com/ePages-de/epages-rest-ruby).
+On the charts side there are lots of options to choose, but we went with [D3](https://d3js.org/), an open source, very powerful and very customizable option.
+The main reason of picking this was purely educational.
+I love learning new stuff, libraries, languages, paradigms, you name it; and I already used [Highcharts](http://www.highcharts.com/), [Google charts](https://developers.google.com/chart/) and [Chart.js](http://www.chartjs.org/), plus I was really curious about it and my team let me decide {% emoji smiley %}.
+
+[Devise](https://github.com/plataformatec/devise) was our option for user management.
+This gem is easy and simple, so it was perfect to deliver fast (even though in the end everything became more complicated).
+For this simplicity reason we choose [Slim](http://slim-lang.com/), [Sass](http://sass-lang.com/) and [CoffeeScript](http://coffeescript.org/).
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: For this simplicity reason we chose...

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 25]
```diff
@@ -0,0 +1,109 @@
...
+
+Join us on our next blog post, where we will talk about the issues we encountered during the development and how everything was about get more complex than we expected.
+
+Thanks for reading and keep coding! :)
```
Comment (BirgitBader) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: ... {% emoji wink %}

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 6 above)
Comment (unaitxu) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Done

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 4 above)
Comment (unaitxu) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Done

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 9 above)
Comment (unaitxu) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Done

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 10 above)
Comment (unaitxu) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Done

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 11 above)
Comment (unaitxu) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Done

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 12 above)
Comment (unaitxu) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Done

Comment (unaitxu) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Done

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 13 above)
Comment (unaitxu) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Done

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 14 above)
Comment (unaitxu) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Done

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 15 above)
Comment (unaitxu) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Done

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 16 above)
Comment (unaitxu) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Done

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 17 above)
Comment (unaitxu) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Done

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 18 above)
Comment (unaitxu) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Done

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 19 above)
Comment (unaitxu) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Done

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 20 above)
Comment (unaitxu) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Done

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 21 above)
Comment (unaitxu) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Done

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 22 above)
Comment (unaitxu) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Done

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 23 above)
Comment (unaitxu) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Done

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 24 above)
Comment (unaitxu) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Done

File: _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 25 above)
Comment (unaitxu) [inline _posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md]: That emoji is not working, I changed to :blush: is it ok?

File: _posts/2017/2017-03-02-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 26]
```diff
@@ -50,7 +50,7 @@ so you have more control on the call and it gets easier to do batch operations.
 We also fixed some bugs and updated some dependencies.
 
 Testing things in the console was fun, but we needed a real life scenario to check that the gem was working correctly and to find missing features.
-Lucky for us, our Product Owner came up with task perfect for this situation.
+Lucky for us, our Product Owner came up with task perfect for this.
```
Comment (BirgitBader) [inline _posts/2017/2017-03-02-creating-a-sales-analytics-app-with-ruby-on-rails.md]: with a perfect task for this

(add the "a")

File: _posts/2017/2017-03-02-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 27]
```diff
@@ -0,0 +1,109 @@
...
+We needed to create a Sales Analytics App for the ePages [App Store](https://blog.epages.com/us/2016/05/14/the-epages-app-store-everything-for-your-e-commerce-success/).
+
+## The new app in Rails
+
+One of the things that I like the most about working at ePages is that we have lots of colleagues willing to help us.
+We had mockups, designs and texts for the whole Web App, so we could focus on programming.
+
+For simplicity we mimicked more or less the API models and added a bit of Rails Magic so it's even easier to use.
+
+{% image blog/blog-sales-app-model.png auto center %}
```
Comment (BirgitBader) [inline _posts/2017/2017-03-02-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Image is distorted

File: _posts/2017/2017-03-02-creating-a-sales-analytics-app-with-ruby-on-rails.md [hunk 28]
```diff
@@ -0,0 +1,109 @@
...
+The main purpose of the Dashboard was to have all the basic information about a shop at a glance.
+So we decided to include a graph of sales, the amount of sales and orders, as well as the top products, and last orders.
+
+{% imagebasic blog/blog-sales-app-sales.png Sales_view %} image {% endimagebasic %}
+
+Then we included three charts on the Sales section.
+The first is the same as in the Dashboard, the second one represents the orders on the selected time frame and the last is the revenue per order, the calculation between the first two graphs.
+We also added the option to export the data to an `xlsx` file so the merchant doesn't need an internet connection to review the sales.
+
+{% imagebasic blog/blog-sales-app-customers.png Customers_view %} image {% endimagebasic %}
```
Comment (BirgitBader) [inline _posts/2017/2017-03-02-creating-a-sales-analytics-app-with-ruby-on-rails.md]: The image preview behaves strangely.

File: _posts/2017/2017-03-02-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 27 above)
Comment (unaitxu) [inline _posts/2017/2017-03-02-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Fixed

File: _posts/2017/2017-03-02-creating-a-sales-analytics-app-with-ruby-on-rails.md (hunk 26 above)
Comment (unaitxu) [inline _posts/2017/2017-03-02-creating-a-sales-analytics-app-with-ruby-on-rails.md]: Done
